# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении.

Выполнили студенты гр. 2381 Долотов Никита и Зазуля Илья. Вариант №18

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальная оценка** – оценка неизвестного параметра $\theta$ в виде интервала $(\hat{\theta}_1;\, \hat{\theta}_2)$, который с заданной вероятностью накрывает истинное значение параметра:
$$
P\!\left(\hat{\theta}_1 < \theta < \hat{\theta}_2\right) = \gamma
$$

**Надёжность (доверительная вероятность)** $\gamma$ – вероятность того, что построенный доверительный интервал содержит истинное значение параметра. Типичные значения: $\gamma = 0.95$ и $\gamma = 0.99$.

**Точность оценки** $\Delta$ – полуширина доверительного интервала:
$$
\Delta = \hat{\theta}_2 - \hat{\theta}_1 \Bigl/ 2, \quad \text{т.е.}\quad \hat{\theta}_{1,2} = \hat{\theta} \mp \Delta
$$

**Уровень значимости** $\alpha = 1 - \gamma$ – вероятность ошибочно отвергнуть верную гипотезу.

---

### Доверительный интервал для математического ожидания

При неизвестном генеральном СКО используется **распределение Стьюдента** с $n-1$ степенями свободы:
$$
\bar{x} - \frac{t_\gamma \cdot s}{\sqrt{n}} < \mu < \bar{x} + \frac{t_\gamma \cdot s}{\sqrt{n}}
$$
где $s$ – исправленное СКО, $t_\gamma = S^{-1}\!\left(\dfrac{1+\gamma}{2},\; n-1\right)$ – квантиль Стьюдента.

### Доверительный интервал для среднеквадратичного отклонения

Строится на основе **хи-квадрат распределения** с $n-1$ степенями свободы:
$$
s\sqrt{\frac{n-1}{\chi^2_{1-\alpha/2}}} < \sigma < s\sqrt{\frac{n-1}{\chi^2_{\alpha/2}}}
$$
где $\chi^2_{\alpha/2}$ и $\chi^2_{1-\alpha/2}$ – квантили $\chi^2$-распределения.

---

### Критерий согласия Пирсона $\chi^2$

Используется для проверки гипотезы $H_0$ о нормальности распределения.

**Наблюдаемое значение критерия:**
$$
\chi^2_\text{набл} = \sum_{i=1}^{k} \frac{(n_i - n'_i)^2}{n'_i} = \sum_{i=1}^{k} \frac{n_i^2}{n'_i} - n
$$
где $n_i$ – эмпирические частоты, $n'_i = n \cdot p_i$ – теоретические частоты.

**Число степеней свободы:** $df = k - r - 1$, где $r = 2$ – число оценённых параметров (для нормального распределения: $\bar{x}$ и $s$).

**Правило принятия решения:**
$$
\chi^2_\text{набл} \leq \chi^2_\text{крит}(\alpha,\, df) \Rightarrow H_0 \text{ не отвергается}
$$
$$
\chi^2_\text{набл} > \chi^2_\text{крит}(\alpha,\, df) \Rightarrow H_0 \text{ отвергается}
$$

---

### Ошибки при проверке гипотез

| Ошибка | Суть | Вероятность |
|--------|------|-------------|
| **Первого рода** | Отвергнуть верную $H_0$. | $\alpha$ |
| **Второго рода** | Принять ложную $H_0$ | $\beta$ |

**Мощность критерия** $= 1 - \beta$ – вероятность правильно отвергнуть ложную гипотезу.


## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы

### Задание 1

Используя значения $\bar{x}$, $s$ и $n$ из практической работы №2,
вычислим квантиль Стьюдента $t_{\alpha/2,\,n-1}$, точность $\Delta$
и границы доверительного интервала для $\mu$ при $\gamma \in \{0.95;\; 0.99\}$.

In [23]:
import numpy as np
from scipy import stats as st

x_mean = 459.69125
s = 52.28756771219408
n = 104
df = n - 1

print(f"Объём выборки: n = {n}")
print(f"Выборочное среднее: x̄ = {x_mean:.4f}")
print(f"Исправленное СКО: s = {s:.4f}\n")

for gamma in [0.95, 0.99]:
    alpha  = 1 - gamma
    t_crit = st.t.ppf(1 - alpha / 2, df)
    delta  = t_crit * s / np.sqrt(n)
    lower  = x_mean - delta
    upper  = x_mean + delta

    print(f"γ = {gamma}  (α = {alpha:.2f})")
    print(f"  Квантиль Стьюдента: t = {t_crit:.4f}")
    print(f"  Точность оценки: Δ = {delta:.4f}")
    print(f"  Доверительный интервал: ({lower:.4f}; {upper:.4f})")
    print(f"  Ширина интервала: 2Δ = {2*delta:.4f}\n")

Объём выборки: n = 104
Выборочное среднее: x̄ = 459.6913
Исправленное СКО: s = 52.2876

γ = 0.95  (α = 0.05)
  Квантиль Стьюдента: t = 1.9833
  Точность оценки: Δ = 10.1686
  Доверительный интервал: (449.5226; 469.8599)
  Ширина интервала: 2Δ = 20.3373

γ = 0.99  (α = 0.01)
  Квантиль Стьюдента: t = 2.6244
  Точность оценки: Δ = 13.4559
  Доверительный интервал: (446.2353; 473.1472)
  Ширина интервала: 2Δ = 26.9118



#### Вывод

| $\gamma$ | $\alpha$ | $t_{\alpha/2,\,103}$ | $\Delta$ | Доверительный интервал | Ширина $2\Delta$ |
|:--------:|:--------:|:--------------------:|:--------:|:----------------------:|:---------------:|
| 0.95 | 0.05 | 1.9833 | 10.1686 | (449.5226; 469.8599) | 20.3373 |
| 0.99 | 0.01 | 2.6244 | 13.4559 | (446.2353; 473.1472) | 26.9118 |

При $\gamma = 0.95$ математическое ожидание генеральной совокупности с надёжностью 95%
находится в интервале $(449.52;\;469.86)$.

При повышении надёжности до $\gamma = 0.99$ интервал расширяется до $(446.24;\;473.15)$:
ширина возрастает с $20.34$ до $26.91$. Это отражает фундаментальный компромисс
математической статистики – повышение надёжности оценки достигается ценой снижения её точности.

### Задание 2

Доверительный интервал для $\sigma$ строится через замену $q = \delta/s$, откуда:
$$
s(1-q) < \sigma < s(1+q)
$$

Значение $q$ определяется из условия на $\chi$-распределение:
$$
\int\limits_{\tfrac{\sqrt{n-1}}{1+q}}^{\tfrac{\sqrt{n-1}}{1-q}} f(\chi,\, n-1)\, d\chi = \gamma
$$

Эквивалентно, через квантили $\chi^2$-распределения с $df = n-1$ степенями свободы:
$$
q_1 = 1 - \sqrt{\frac{n-1}{\chi^2_{1-\alpha/2,\,n-1}}}, \quad
q_2 = \sqrt{\frac{n-1}{\chi^2_{\alpha/2,\,n-1}}} - 1
$$
Так как $\chi^2$-распределение асимметрично, $q_1 \neq q_2$ – интервал несимметричен относительно $s$.

In [24]:
print(f"n = {n},  s = {s:.4f}\n")

for gamma in [0.95, 0.99]:
    alpha = 1 - gamma
    chi2_lower = st.chi2.ppf(alpha / 2, df)
    chi2_upper = st.chi2.ppf(1 - alpha / 2, df)

    sigma_lower = s * np.sqrt(df / chi2_upper)
    sigma_upper = s * np.sqrt(df / chi2_lower)

    q1 = 1 - np.sqrt(df / chi2_upper)
    q2 = np.sqrt(df / chi2_lower) - 1

    print(f"γ = {gamma}  (α = {alpha:.2f})")
    print(f"  χ²_крит: [{chi2_lower:.3f};  {chi2_upper:.3f}]")
    print(f"  q: [{q1:.4f};  {q2:.4f}]")
    print(f"  Доверительный интервал для σ: ({sigma_lower:.4f};  {sigma_upper:.4f})")
    print(f"  Ширина интервала: {sigma_upper - sigma_lower:.4f}\n")

n = 104,  s = 52.2876

γ = 0.95  (α = 0.05)
  χ²_крит: [76.809;  132.975]
  q: [0.1199;  0.1580]
  Доверительный интервал для σ: (46.0185;  60.5497)
  Ширина интервала: 14.5312

γ = 0.99  (α = 0.01)
  χ²_крит: [69.785;  143.712]
  q: [0.1534;  0.2149]
  Доверительный интервал для σ: (44.2660;  63.5236)
  Ширина интервала: 19.2576



#### Вывод

| $\gamma$ | $\chi^2_{\alpha/2}$ | $\chi^2_{1-\alpha/2}$ | $q_1$ | $q_2$ | Доверительный интервал | Ширина |
|:--------:|:-------------------:|:---------------------:|:-----:|:-----:|:----------------------:|:------:|
| 0.95 | 76.809 | 132.975 | 0.1199 | 0.1580 | (46.0185; 60.5497) | 14.5312 |
| 0.99 | 69.785 | 143.712 | 0.1534 | 0.2149 | (44.2660; 63.5236) | 19.2576 |

Точечная оценка $s = 52.29$ не является серединой доверительного интервала –
следствие асимметрии $\chi^2$-распределения, из-за которой $q_1 \neq q_2$.

С ростом надёжности $\gamma$ интервал расширяется: ширина увеличивается с $14.53$
до $19.26$. Это согласуется с общим принципом: более высокая гарантия накрытия
истинного $\sigma$ достигается ценой снижения точности оценки.

Построенные интервалы справедливы при выполнении предположения о нормальности
генеральной совокупности.

### Задание 3

Гипотезы:
$$H_0: \text{генеральная совокупность имеет нормальное распределение } N(\mu, \sigma^2)$$
$$H_1: \text{распределение отличается от нормального}$$

Теоретическая вероятность попадания в $i$-й интервал $[a_i;\, a_{i+1})$:
$$p_i = \Phi\!\left(\frac{a_{i+1} - \bar{x}}{s}\right) - \Phi\!\left(\frac{a_i - \bar{x}}{s}\right)$$

Теоретическая частота: $n'_i = n \cdot p_i$

In [25]:
import pandas as pd
from scipy.stats import norm


x_mids = np.array([361.44, 396.31, 431.19, 466.06, 500.94, 535.81, 570.69, 605.56])
n_i = np.array([3, 17, 29, 22, 20, 8, 3, 2], dtype=float)
h = 34.88
k = len(n_i)

left_bounds  = x_mids - h / 2
right_bounds = x_mids + h / 2
left_bounds[0]   = -np.inf
right_bounds[-1] =  np.inf

p_i = norm.cdf(right_bounds, x_mean, s) - norm.cdf(left_bounds, x_mean, s)
p_i = p_i / p_i.sum()
n_prime = n * p_i

diff2 = (n_i - n_prime) ** 2
diff2_div = diff2 / n_prime
ni2 = n_i ** 2
ni2_div = ni2 / n_prime

interval_labels = []
for i in range(k):
    l = "-∞" if np.isinf(left_bounds[i]) else f"{left_bounds[i]:.2f}"
    r = "+∞" if np.isinf(right_bounds[i]) else f"{right_bounds[i]:.2f}"
    interval_labels.append(f"[{l}, {r})")

df_table = pd.DataFrame({
    'i': list(range(1, k + 1)),
    '[x_i, x_{i+1})': interval_labels,
    'n_i': n_i,
    'p_i': np.round(p_i, 6),
    "n'_i": np.round(n_prime, 4),
    "(n_i-n'_i)²": np.round(diff2, 4),
    "(n_i-n'_i)²/n'_i": np.round(diff2_div, 4),
    "n²_i": ni2,
    "n²_i/n'_i": np.round(ni2_div, 4),
})

sum_row = pd.DataFrame([{
    'i': 'Σ',
    '[x_i, x_{i+1})': '—',
    'n_i': n_i.sum(),
    'p_i': round(p_i.sum(), 6),
    "n'_i": round(n_prime.sum(), 4),
    "(n_i-n'_i)²": '—',
    "(n_i-n'_i)²/n'_i": round(diff2_div.sum(), 4),
    "n²_i": '—',
    "n²_i/n'_i": round(ni2_div.sum(), 4),
}])

df_table = pd.concat([df_table, sum_row], ignore_index=True)
df_table

,i,"[x_i, x_{i+1})",n_i,p_i,n'_i,(n_i-n'_i)²,(n_i-n'_i)²/n'_i,n²_i,n²_i/n'_i
0,1,"[-∞, 378.88)",3.0,0.061102,6.3546,11.2535,1.7709,9.0,1.4163
1,2,"[378.87, 413.75)",17.0,0.128696,13.3844,13.0728,0.9767,289.0,21.5923
2,3,"[413.75, 448.63)",29.0,0.226396,23.5452,29.7546,1.2637,841.0,35.7185
3,4,"[448.62, 483.50)",22.0,0.259376,26.9751,24.7515,0.9176,484.0,17.9425
4,5,"[483.50, 518.38)",20.0,0.193563,20.1305,0.017,0.0008,400.0,19.8703
5,6,"[518.37, 553.25)",8.0,0.094086,9.7850,3.1862,0.3256,64.0,6.5406
6,7,"[553.25, 588.13)",3.0,0.029761,3.0952,0.0091,0.0029,9.0,2.9078
7,8,"[588.12, +∞)",2.0,0.007020,0.7301,1.6128,2.2091,4.0,5.4791
8,Σ,—,104.0,1.000000,104.0000,—,7.4674,—,111.4674


#### Вывод

По выборочным оценкам $\bar{x} = 459.69$ и $s = 52.29$ для каждого из $k = 8$
интервалов вычислены теоретические вероятности $p_i$ и частоты $n'_i = n \cdot p_i$.
Наблюдаемое значение критерия: $\chi^2_\text{набл} = 7.4674$.

### Задание 4

Покажем, что основная формула критерия Пирсона эквивалентна упрощённой:

$$
\chi^2_{\text{набл}} = \sum_{i=1}^k \frac{(n_i - n'_i)^2}{n'_i} = \sum_{i=1}^k \frac{n_i^2}{n'_i} - n
$$

Раскроем числитель:

$$
\sum_{i=1}^k \frac{(n_i - n'_i)^2}{n'_i} =
\sum_{i=1}^k \frac{n_i^2 - 2n_i n'_i + n_i'^2}{n'_i} =
\sum_{i=1}^k \frac{n_i^2}{n'_i} - 2\sum_{i=1}^k n_i + \sum_{i=1}^k n'_i
$$

Поскольку сумма эмпирических и теоретических частот равна объёму выборки:

$$
\sum_{i=1}^k n_i = n, \qquad \sum_{i=1}^k n'_i = n
$$

Подставляем:

$$
\sum_{i=1}^k \frac{n_i^2}{n'_i} - 2n + n = \sum_{i=1}^k \frac{n_i^2}{n'_i} - n \qquad \blacksquare
$$


In [28]:
chi2_formula1 = diff2_div.sum()
chi2_formula2 = ni2_div.sum() - n

print(f"χ²_набл по основной формуле: {chi2_formula1:.4f}")
print(f"χ²_набл по эквивалентной формуле: {chi2_formula2:.4f}")

if np.isclose(chi2_formula1, chi2_formula2):
    print("Результаты совпадают - вычисления корректны.")
else:
    print("Расхождение в вычислениях.")

χ²_набл по основной формуле: 7.4674
χ²_набл по эквивалентной формуле: 7.4674
Результаты совпадают - вычисления корректны.


#### Вывод

Алгебраически доказано, что формулы

$$\sum_{i=1}^k \frac{(n_i - n'_i)^2}{n'_i} \quad \text{и} \quad \sum_{i=1}^k \frac{n_i^2}{n'_i} - n$$

тождественно равны, что следует из свойств $\sum n_i = \sum n'_i = n$.

Численная проверка подтверждает корректность вычислений: оба выражения дают
$\chi^2_\text{набл} = 7.4674$.

### Задание 5

Число степеней свободы: $df = k - 3 = 8 - 3 = 5$ (вычитаем 1 и два оценённых параметра $\bar{x}$, $s$).

Критическая область – правосторонняя: $H_0$ отвергается при $\chi^2_\text{набл} > \chi^2_\text{крит}(\alpha, df)$.

In [29]:
alpha = 0.05
df_chi = k - 3
chi2_crit = st.chi2.ppf(1 - alpha, df_chi)
chi2_obs = diff2_div.sum()

print(f"Уровень значимости: α = {alpha}")
print(f"Число интервалов: k = {k}")
print(f"Число степеней свободы: df = k - 3 = {df_chi}")
print(f"Критическое значение: χ²_крит(0.05, {df_chi}) = {chi2_crit:.4f}")
print(f"Наблюдаемое значение: χ²_набл = {chi2_obs:.4f}")
print()
if chi2_obs < chi2_crit:
    print(f"χ²_набл = {chi2_obs:.4f} < χ²_крит = {chi2_crit:.4f} => H₀ не отвергается")
else:
    print(f"χ²_набл = {chi2_obs:.4f} >= χ²_крит = {chi2_crit:.4f} => H₀ отвергается")

Уровень значимости: α = 0.05
Число интервалов: k = 8
Число степеней свободы: df = k - 3 = 5
Критическое значение: χ²_крит(0.05, 5) = 11.0705
Наблюдаемое значение: χ²_набл = 7.4674

χ²_набл = 7.4674 < χ²_крит = 11.0705 => H₀ не отвергается


#### Вывод

| Параметр | Значение |
|----------|----------|
| Уровень значимости $\alpha$ | 0.05 |
| Число степеней свободы $df$ | 5 |
| $\chi^2_\text{крит}(0.05,\; 5)$ | 11.0705 |
| $\chi^2_\text{набл}$ | 7.4674 |

$$\chi^2_\text{набл} = 7.4674 < \chi^2_\text{крит} = 11.0705 \quad \Rightarrow \quad H_0 \text{ не отвергается}$$

На уровне значимости $\alpha = 0.05$ нет оснований отвергнуть гипотезу о нормальном
распределении генеральной совокупности. Расхождения между эмпирическими и теоретическими
частотами статистически незначимы – выборка согласуется с $N(\mu, \sigma^2)$.

## Выводы

В ходе практической работы получены навыки построения интервальных оценок параметров
распределения и проверки статистических гипотез о нормальности.

Доверительные интервалы для математического ожидания построены на основе
$t$-распределения Стьюдента с $df = 103$ степенями свободы:

| $\gamma$ | Доверительный интервал для $\mu$ |
|:--------:|:--------------------------------:|
| 0.95 | $(449.52;\; 469.86)$ |
| 0.99 | $(446.24;\; 473.15)$ |

Доверительные интервалы для среднеквадратичного отклонения построены на основе
$\chi^2$-распределения:

| $\gamma$ | Доверительный интервал для $\sigma$. |
|:--------:|:-----------------------------------:|
| 0.95 | $(46.02;\; 60.55)$ |
| 0.99 | $(44.27;\; 63.52)$ |

В обоих случаях с ростом надёжности $\gamma$ интервалы расширяются, что отражает
компромисс между точностью и надёжностью оценки.

Проверка гипотезы о нормальности критерием Пирсона при $\alpha = 0.05$ и $df = 5$
дала:
$$\chi^2_\text{набл} = 7.4674 < \chi^2_\text{крит} = 11.0705 \quad \Rightarrow \quad H_0 \text{ не отвергается}$$

Расхождения между эмпирическими и теоретическими частотами статистически незначимы.
Выборка согласуется с нормальным распределением $N(\mu, \sigma^2)$, что обосновывает корректность применения $t$-распределения и $\chi^2$-распределения
при построении доверительных интервалов.